lib/ansible/modules/windows/win_psmodule.ps1

```diff
@@ -1,186 +1,380 @@
+        $LatestModuleVersion = $LatestModule.Version
+    }
+    catch {
+        $ErrorMessage = "Cant find the module $($Name): $($_.Exception.Message)"
         Fail-Json $result $ErrorMessage
-      }
     }
+
+    $LatestModuleVersion
 }
 
-Function Remove-PsModule {
-    param(
-      [Parameter(Mandatory=$true)]
-      [string]$Name,
-      [bool]$CheckMode
-    )
-    # If module is present, unistalls it.
-    if (Get-Module -Listavailable|?{$_.name -eq $Name}){
-      try{
-        Uninstall-Module -Name $Name -Confirm:$false -Force -ErrorAction Stop -WhatIf:$CheckMode | out-null
-        $result.output = "Module $($Name) removed"
-        $result.changed = $true
-      }
-      catch{
-        $ErrorMessage = "Problems removing $($Name) module: $($_.Exception.Message)"
+# Check PowerShell version, fail if < 5.0
+$PsVersion = $PSVersionTable.PSVersion
+if ($PsVersion.Major -lt 5 ){
+    $ErrorMessage = "Windows PowerShell 5.0 or higher is needed"
+    Fail-Json $result $ErrorMessage
+}
+
+if ( $required_version -and ( $minimum_version -or $maximum_version ) ) {
+       $ErrorMessage = "Parameters required_version and minimum/maximum_version are mutually exclusive."
+       Fail-Json $result $ErrorMessage
+}
+
+if ( $allow_prerelease -and ( $minimum_version -or $maximum_version ) ) {
+    $ErrorMessage = "Parameters minimum_version, maximum_version can't be used with the parameter allow_prerelease."
+    Fail-Json $result $ErrorMessage
+}
+
+if ( $allow_prerelease -and $state -eq "absent" ) {
+    $ErrorMessage = "The parameter allow_prerelease can't be used with state set to 'absent'."
+    Fail-Json $result $ErrorMessage
+}
+
+if ( ($state -eq "latest") -and
+    ( $required_version -or $minimum_version -or $maximum_version ) ) {
+        $ErrorMessage = "When the parameter state is equal 'latest' you can use any of required_version, minimum_version, maximum_version."
         Fail-Json $result $ErrorMessage
-      }
+}
 
+if ( $repo ) {
+    $RepositoryExists = Get-PSRepository -Name $repo -ErrorAction Ignore
+    if ( $null -eq $RepositoryExists) {
+        $ErrorMessage = "The repository $repo doesn't exist."
+        Fail-Json $result $ErrorMessage
     }
-    else{
-      $result.output = "Module $($Name) not present"
-    }
+
 }
 
-# Check powershell version, fail if < 5.0
-$PsVersion = $PSVersionTable.PSVersion
-if ($PsVersion.Major -lt 5){
-  $ErrorMessage = "Powershell 5.0 or higher is needed"
-  Fail-Json $result $ErrorMessage
+if ( ($allow_clobber -or $allow_prerelease -or
+    $required_version -or $minimum_version -or $maximum_version) ) {
+    # Update the PowerShellGet and PackageManagement modules.
+    # It's required to support AllowClobber, AllowPrerelease parameters.
+    Install-PrereqModule -CheckMode $check_mode
 }
 
 if ($state -eq "present") {
-    if (($repo) -and ($url)) {
-        Install-Repository -Name $repo -Url $url -CheckMode $check_mode 
+    if ($name) {
+        $ht = @{
```

Comment:
_sigh: I wish we could do inline hashtables instead of needing to create a variable_

Reply:
I didn't get that.

Reply:
You can just pass the parameters instead of creating a new hash table for that specific purpose.